catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/web/rest/IcebergTableOperations.java

```diff
@@ -158,6 +159,23 @@ public Response tableExists(
     }
   }
 
+  @GET
+  @Path("{table}/partitions")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Timed(name = "load-table-partitions." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "load-table-partitions", absolute = true)
+  public Response loadTablePartitions(
+      @PathParam("namespace") String namespace,
+      @PathParam("table") String table,
+      @DefaultValue("all") @QueryParam("partitions-filter") String partitionsFilter,
+      @DefaultValue("all") @QueryParam("snapshots") String snapshots) {
+    // todo support snapshots
+    TableIdentifier tableIdentifier =
+        TableIdentifier.of(RESTUtil.decodeNamespace(namespace), table);
+    ListPartitionsResponse listPartitionsResponse = icebergTableOps.listPartitions(tableIdentifier);
+    return IcebergRestUtils.ok(listPartitionsResponse);
+  }
```

Comment:
Hi @SinghAsDev thanks a lot for your contribution. Is this a newly added Iceberg REST spec, or it is just an extension added by us? CC @FANNG1

Reply:
it's not the Iceberg REST spec, but it seems to be a general requirement